### ISSUE TITLE
Add Bundler cop to discourage ifs vs install_if

### ISSUE THREAD

**Issue by dalexj** [rubocop/rubocop#12093]
## Is your feature request related to a problem? Please describe.
The cop for `Bundler/DuplicatedGem` recommends the following treatment to avoid specifying a gem multiple times:
```ruby
# good - conditional declaration
if Dir.exist?(local)
  gem 'rubocop', path: local
elsif ENV['RUBOCOP_VERSION'] == 'master'
  gem 'rubocop', git: 'https://github.com/rubocop/rubocop.git'
else
  gem 'rubocop', '~> 0.90.0'
end
```
Using `if` is dangerous as specified gems wont be included in your lockfile due to your environment never seeing a gem exists, and updates won't apply. Bundler recommends using the `install_if` method for these purposes https://bundler.io/v2.4/man/gemfile.5.html#INSTALL_IF

## Describe the solution you'd like

- The docs for `Bundler/DuplicatedGem` should not encourage using `if` for specifying gems.
- Adding a new cop to warn of potential issues of using `if`.

it looks like `install_if` was added in bundler `1.10.0` (2015 - https://github.com/rubygems/bundler/blob/master/CHANGELOG.md#1100pre-2015-05-03). I don't know if there is a way to target bundler versions for cops

## Describe alternatives you've considered

I could be wrong and there is a good way to specify multiple entries or different gems using `if`

**Comment by koic** [comment]
Interesting suggestion. I'm not familiar with `install_if`, what would be `good` example for the `# good - conditional declaration` example? I would like to know how a conditional branch using `else` is rewritten by `install_if`.
